dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -354,10 +365,46 @@ def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
             logger.info(f"Will truncate tables {truncate_tables}")
             self.truncate_tables(list(truncate_tables))
 
+        # if the init file already exists, we return
+        if self.fs_client.exists(self.pathlib.join(self.dataset_path, INIT_FILE_NAME)):
+            return
+
         # we mark the storage folder as initialized
         self.fs_client.makedirs(self.dataset_path, exist_ok=True)
         self.fs_client.touch(self.pathlib.join(self.dataset_path, INIT_FILE_NAME))
 
+        # we set the version to "2" to indicate that .gz extension is added by default
+        # version "1" corresponds to an empty init file and indicates that .gz is not added to compressed files
+        self.fs_client.write_text(
+            self.pathlib.join(self.dataset_path, INIT_FILE_NAME),
+            json.dumps({"version": 2}),
+            encoding="utf-8",
+        )
+
+    def get_storage_version(self) -> int:
```

Comment:
This is very good! We just need to cache the version on the client which should be easy. it will not change during the life of the client.